androidgen/src/main/java/com/azure/autorest/android/mapper/AndroidOptionalParameterMapper.java

```diff
@@ -0,0 +1,128 @@
+package com.azure.autorest.android.mapper;
+
+import com.azure.autorest.extension.base.model.codemodel.Parameter;
+import com.azure.autorest.mapper.Mappers;
+import com.azure.autorest.model.clientmodel.ClientMethodParameter;
+import com.azure.autorest.model.clientmodel.ClientModel;
+import com.azure.autorest.model.clientmodel.ClientModelProperty;
+import com.azure.autorest.model.clientmodel.IType;
+import com.azure.autorest.util.CodeNamer;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class AndroidOptionalParameterMapper {
+    private String packageName;
+    private String methodName;
+    private List<Parameter> parameters = new ArrayList<Parameter>();
+
+    public AndroidOptionalParameterMapper packageName(String packageName) {
+        this.packageName = packageName;
+        return this;
+    }
+    public AndroidOptionalParameterMapper methodName(String methodName) {
+        this.methodName = methodName;
+        return this;
+    }
+
+    public AndroidOptionalParameterMapper parameters(List<Parameter> parameters) {
+        this.parameters = parameters;
+        return this;
+    }
+
+    public boolean hasOptionalParameters() {
+        return parameters.size() > 0;
+    }
+
+    public String getModelTypeName() {
+        return CodeNamer.toPascalCase(String.format("%1$sOptions", methodName));
```

Comment:
Will it happen that 2 method in different operation group get same name (likely for `list` etc.), which result in same model name for Options class?

Reply:
Valid concern but that's kind of the assumption we made above this level already. For example, if two operations at two different paths have the same "operationId", they will cause name clash in the client (even though may have different signatures). We can work around that by modifying the swagger file so that they have different operationId. Fixing this assumption in autorest requires a lot more effort than we can afford. I'll just take a note and investigate later.